agent/agent.go

```diff
@@ -101,26 +123,51 @@ func DefaultConfig() (Config, error) {
 	}, nil
 }
 
-// Start the Agent which includes the registration against Consul Agent
-func (a *Agent) Start() error {
-	log.Println("Registering the agent service with Consul...")
-	err := a.registerConsulService()
+func DefaultAgentConfig() (Config, error) {
+	hostname, err := os.Hostname()
 	if err != nil {
-		return errors.Wrap(err, "could not register consul service")
+		return Config{}, errors.Wrap(err, "could not read the hostname")
 	}
-	log.Println("Consul service registered.")
 
-	defer func() {
-		log.Println("De-registering the agent service with Consul...")
-		err := a.consul.Agent().ServiceDeregister(a.cfg.InstanceName)
+	return Config{
+		InstanceName: hostname,
+		DiscoveryTTL: 15 * time.Second,
+		CheckerTTL:   10 * time.Second,
+	}, nil
+}
+
+// Start the Agent which includes the registration against Consul Agent
+func (a *Agent) Start() error {
+	status := make(chan int, 1)
+	var wg sync.WaitGroup
+
+	// Attempt to register against the consul client
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		log.Println("Registering the agent service with Consul...")
+		defer wg.Done()
+		err := a.startRegisterLoop()
 		if err != nil {
-			log.Println("An error occurred while trying to deregisterConsulService the agent service with Consul:", err)
+			log.Fatal("Error while registering consul")
 		} else {
-			log.Println("Consul service de-registered.")
+			status <- 1
```

Comment:
I'm not sure we need this channel at all.

Reply:
It was supposed to be used on `embeddedconsul.go:98` (the `ServiceDeregister` call). In the end, it's working fine without checking if the service actually needs de-register before deregistering it, but I wanted to be sure before removing it. I'll re-test and ensure we can remove it.